### DISCUSSION TITLE
How to close the tcp frame stream from client.

### DISCUSSION THREAD

**Discussion by fy2462** [tokio-rs/tokio#4281] [Q&A] [answered]
Hi guys, 

I want to close the frame stream from the client, and the server-side can get the stream close event or exception.

How to do that, any help?

Thanks very much.

**Comment by Darksonn** [comment, marked as answer]
Typically you would use either [`AsyncWriteExt::shutdown`][1] or [`SinkExt::close`][2] depending on the type of IO resources.

[1]: https://docs.rs/tokio/1/tokio/io/trait.AsyncWriteExt.html#method.shutdown
[2]: https://docs.rs/futures/0.3/futures/sink/trait.SinkExt.html#method.close

**Reply by fy2462** [reply]
Thanks a lot, I have got the close event from server-side after calling the `shutdown`.  @Darksonn
